cmake/DaemonFlags.cmake

```diff
@@ -201,256 +184,350 @@ if (MSVC)
-
-                find_program(GCC_RANLIB
-                    NAMES
-                        "${TOOLCHAIN_PREFIX}gcc-ranlib"
-                        "${TOOLCHAIN_PREFIX}gcc-ranlib-${_version}"
-                    DOC "gcc provided wrapper for ranlib which adds the --plugin option"
-                )
-
-                mark_as_advanced(GCC_AR GCC_RANLIB)
-
-                # Override standard ar and ranlib with the gcc- versions
-                if (GCC_AR)
-                    set(CMAKE_AR ${GCC_AR})
-                endif()
-                if (GCC_RANLIB)
-                    set(CMAKE_RANLIB ${GCC_RANLIB})
-                endif()
-
-                try_c_cxx_flag(NO_FAT_LTO_OBJECTS "-fno-fat-lto-objects")
-            else()
-                try_c_cxx_flag(FAT_LTO_OBJECTS "-ffat-lto-objects")
-            endif()
-        endif()
+        set_c_cxx_flag("/GL" MINSIZEREL)
+        set_c_cxx_flag("/GL" RELWITHDEBINFO)
+        set_c_cxx_flag("/GL" RELEASE)
+        set_linker_flag("/LTCG" MINSIZEREL)
+        set_linker_flag("/LTCG" RELWITHDEBINFO)
+        set_linker_flag("/LTCG" RELEASE)
     endif()
-
+else()
+	# Minimum language standards supported.
+
+	option(USE_RECOMMENDED_C_STANDARD "Use recommended C standard" ON)
+	mark_as_advanced(USE_RECOMMENDED_C_STANDARD)
+
+	if (USE_RECOMMENDED_C_STANDARD)
+		# Required when building gzip or it would complain about
+		# implicitly defined lseek, read, write and close.
+		try_c_flag(GNU89 "-std=gnu89")
+		if (NOT FLAG_GNU89)
+			set(NOT_GNU_C ON)
+			try_c_flag(C99 "-std=c99")
+			if (NOT FLAG_C99)
+				message(FATAL_ERROR "GNU89 or C99 not supported by compiler")
+			endif()
+		endif()
+	endif()
+
+	option(USE_RECOMMENDED_CXX_STANDARD "Use recommended C++ standard" ON)
+	mark_as_advanced(USE_RECOMMENDED_CXX_STANDARD)
+
+	if (USE_RECOMMENDED_CXX_STANDARD)
+		# PNaCl only defines isascii if __STRICT_ANSI__
+		# is not defined, try GNU variant first.
+		try_cxx_flag(GNUXX14 "-std=gnu++14")
+		if (NOT FLAG_GNUXX14)
+			try_cxx_flag(GNUXX1Y "-std=gnu++1y")
+			if (NOT FLAG_GNUXX1Y)
+				set(NOT_GNU_CXX ON)
+				try_cxx_flag(CXX14 "-std=c++14")
+				if (NOT FLAG_CXX14)
+					try_cxx_flag(CXX1Y "-std=c++1y")
+					if (NOT FLAG_CXX1Y)
+						message(FATAL_ERROR "GNU++14 or C++14 not supported by compiler")
+					endif()
+				endif()
+			endif()
+		endif()
+	endif()
+
+	# If GNU dialects of standards are not found.
+	if (NOT_GNU_C OR NOT_GNU_CXX)
+		message(WARNING "Enabling compiler compatibility macros")
+		# snprintf, vsnprintf:
+		#   _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 500
+		#   || _ISOC99_SOURCE || _BSD_SOURCE
+		# fseeko, ftello:
+		#   _POSIX_C_SOURCE >= 200112L
+		# strdup:
+		#   _POSIX_C_SOURCE >= 200809L || _XOPEN_SOURCE >= 500
+		#   || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED)
+		#   || _BSD_SOURCE || _SVID_SOURCE || _GNU_SOURCE
+		# isascii:
+		#   _XOPEN_SOURCE || _DEFAULT_SOURCE || _SVID_SOURCE
+		add_definitions(-D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=500)
```

Comment:
What platform is this stuff needed on?

Reply:
With `-std=c++14` instead of `-std=gnu++14`.

Reply:
I would understand -DUSE_RECOMMENDED_C_STANDARD=0 as telling the build system to get out of the way and not set *any* C standard flags. I oppose having it mean keep setting language standard flags, but it a different way.

Reply:
The code now only sets GNU dialects.